app/components/login_button_component.rb

```diff
@@ -0,0 +1,29 @@
+class LoginButtonComponent < ButtonComponent
+  VALID_COLORS = ['primary', 'primary-darker', 'primary-lighter'].freeze
+
+  attr_reader :color, :tag_options
+
+  def initialize(color: 'primary', **tag_options)
+    if !VALID_COLORS.include?(color)
+      raise ArgumentError, "`color` #{color}} is invalid, expected one of #{VALID_COLORS}"
+    end
+
+    super(
+      **tag_options
+    )
```

Comment:
Since we're not extending any `ButtonComponent` behavior, we should remove it as the superclass.

```suggestion
class LoginButtonComponent < BaseComponent
  VALID_COLORS = ['primary', 'primary-darker', 'primary-lighter'].freeze

  attr_reader :color, :tag_options

  def initialize(color: 'primary', **tag_options)
    if !VALID_COLORS.include?(color)
      raise ArgumentError, "`color` #{color}} is invalid, expected one of #{VALID_COLORS}"
    end
```